## Day 05 - Exploratory Data Analysis (EDA)

## Objective
Calculate key business metrics and understand overall business performance.

In [1]:
import pandas as pd

df = pd.read_excel("C:/Data Projects/Ecommerce_Sales_Analysis/Data/Superstore.xlsx")

### Calculated Total Sales.

In [2]:
total_sales = df["Sales"].sum()

print(f"Total Sales: {total_sales:,.2f}")

Total Sales: 2,297,200.86


### Calculated Total Profit.

In [3]:
total_profit = df["Profit"].sum()

print(f"Total Profit: {total_profit:,.2f}")

Total Profit: 286,397.02


### Counted the total number of unique orders.

In [8]:
total_orders = df["Order ID"].nunique()

print(f"Total Orders: {total_orders}")

Total Orders: 5009


### Counted the total number of unique customers.

In [9]:
total_customers = df["Customer ID"].nunique()

print(f"Total Customers: {total_customers}")

Total Customers: 793


### Calculated the Average Order Value (AOV).

In [10]:
order_sales = df.groupby("Order ID")["Sales"].sum()

average_order_value = order_sales.mean()

print(f"Average Order Value: {average_order_value:,.2f}")


Average Order Value: 458.61


## Profit Margin

In [11]:
profit_margin = (total_profit / total_sales) * 100

print(f"Profit Margin: {profit_margin:.2f}%")

Profit Margin: 12.47%


### Calculated the Average Orders per Customer.

In [12]:
average_orders = total_orders / total_customers

print(f"Average Orders per Customer: {average_orders:.2f}")

Average Orders per Customer: 6.32


## Key Performance Indicators (KPIs)

| KPI | Value |
|------|-------:|
| Total Sales | 2,297,200.86 |
| Total Profit | 286,397.02 |
| Total Orders | 5,009 |
| Total Customers | 793 |
| Average Order Value | 458.61 |
| Profit Margin | **12.47%** |
| Average Orders per Customer | **6.32** |
---

## Additional Business Metrics

- **Profit Margin:** *(Add your calculated result here after running the code.)*
- **Average Orders per Customer:** *(Add your calculated result here after running the code.)*

---

## Key Insights

- The company generated **$2.30 million** in total sales.
- The business achieved a total profit of **$286,397.02**.
- The overall **profit margin is 12.47%**, indicating that the company earns approximately **$12.47** in profit for every **$100** in sales.
- A total of **793 customers** placed **5,009 unique orders**.
- On average, each customer placed **6.32 orders**, suggesting a good level of customer retention and repeat purchases.
- The average order value is **$458.61**, representing the average revenue generated per order.

---

## Skills Learned

- Calculating business KPIs using Pandas.
- Using `sum()` to calculate totals.
- Using `nunique()` to count unique values.
- Using `groupby()` to aggregate data.
- Calculating Average Order Value (AOV).
- Calculating Profit Margin.
- Calculating customer purchase frequency.
- Interpreting business metrics to support decision-making.

---

## Conclusion

The exploratory analysis provided a high-level overview of the company's performance. The business generated strong sales, maintained a positive profit margin, and demonstrated repeat customer purchases. These KPIs provide a solid foundation for deeper analysis of products, regions, customer segments, and sales trends in the following stages.